astropy/units/tests/test_quantity_non_ufuncs.py

```diff
@@ -2578,3 +2585,144 @@ def test_all_included(self):
     @needs_array_function
     def test_ignored_are_untested(self):
         assert IGNORED_FUNCTIONS | TBD_FUNCTIONS == untested_functions
+
+
+@pytest.mark.parametrize(
+    "target, helper",
+    sorted(
+        itertools.chain(FUNCTION_HELPERS.items(), DISPATCHED_FUNCTIONS.items()),
+        key=lambda items: items[0].__name__,
+    ),
+    ids=lambda func: func.__name__,
+)
+class TestFunctionHelpersSignatureCompatibility:
+    """
+    Check that a helper function's signature is *at least* as flexible
+    as the helped (target) function's. E.g., any argument that is allowed positionally,
+    or as keyword, by the target must be re-exposed *somehow* by the helper.
+    We explicitly allow helper's signature to be *more* flexible than the target signature
+    by allowing *args and **kwargs catch-all arguments, which we use to limit code
+    duplication, and also help with forward and backward compatibility.
+    See https://github.com/astropy/astropy/issues/15703
+    """
+
+    @staticmethod
+    def have_catchall_argument(parameters, kind) -> bool:
+        return any(p.kind is kind for p in parameters.values())
+
+    @staticmethod
+    def get_param_group(parameters, kinds: list) -> list[str]:
+        return [name for name, p in parameters.items() if p.kind in kinds]
+
+    def test_all_arguments_reexposed(self, target, helper):
+        try:
```

Comment:
Well it's certainly possible to filter it with a heuristic that makes sense *now* (for instance we could evaluate `isinstance(obj, types.BuiltinFunctionType)`) but I'm thinking, `inspect.signature` *may* outgrow its current limitations in the future so keeping the `try/except` structure may pay off at some point ?

Reply:
Yes, when I think more about it, it would seem that you would need to construct the list for `parametrize` outside of the class and filter using `try/except` - by that time, it is probably more lines than you have now!  Let's keep it as is for now.